src/algebra/lie/base_change.lean

```diff
@@ -0,0 +1,155 @@
+          linear_map.map_add], }, },
+    { intros y₁ y₂ hy₁ hy₂,
+      simp only [add_add_add_comm, hy₁, hy₂, add_zero, linear_map.add_apply,
+        linear_map.map_add], }, },
+end
+
+private lemma bracket_leibniz_lie (x y z : A ⊗[R] L) : ⁅x, ⁅y, z⁆⁆ = ⁅⁅x, y⁆, z⁆ + ⁅y, ⁅x, z⁆⁆ :=
+begin
+  simp only [bracket_def],
+  apply x.induction_on,
+  { simp only [linear_map.map_zero, add_zero, eq_self_iff_true, linear_map.zero_apply], },
+  { intros a₁ l₁,
+    apply y.induction_on,
+    { simp only [linear_map.map_zero, add_zero, eq_self_iff_true, linear_map.zero_apply], },
+    { intros a₂ l₂,
+      apply z.induction_on,
+      { simp only [linear_map.map_zero, add_zero], },
+      { intros a₃ l₃, simp only [bracket'_tmul],
+        rw [mul_left_comm a₂ a₁ a₃, mul_assoc, leibniz_lie, tensor_product.tmul_add], },
+      { intros u₁ u₂ h₁ h₂,
+        simp only [add_add_add_comm, h₁, h₂, linear_map.map_add], }, },
+    { intros u₁ u₂ h₁ h₂,
+      simp only [add_add_add_comm, h₁, h₂, linear_map.add_apply, linear_map.map_add], }, },
+  { intros u₁ u₂ h₁ h₂,
+    simp only [add_add_add_comm, h₁, h₂, linear_map.add_apply, linear_map.map_add], },
+end
+
+instance : lie_ring (A ⊗[R] L) :=
+{ add_lie     := λ x y z, by simp only [bracket_def, linear_map.add_apply, linear_map.map_add],
+  lie_add     := λ x y z, by simp only [bracket_def, linear_map.map_add],
+  lie_self    := bracket_lie_self R A L,
+  leibniz_lie := bracket_leibniz_lie R A L, }
+
+private lemma bracket_lie_smul (a : A) (x y : A ⊗[R] L) : ⁅x, a • y⁆ = a • ⁅x, y⁆ :=
+begin
+  apply x.induction_on,
+  { simp only [zero_lie, smul_zero], },
+  { intros a₁ l₁, apply y.induction_on,
+    { simp only [lie_zero, smul_zero], },
+    { intros a₂ l₂,
+      simp only [bracket_def, bracket', tensor_product.smul_tmul', mul_left_comm a₁ a a₂,
+        tensor_product.curry_apply, algebra.lmul'_apply, algebra.id.smul_eq_mul, function.comp_app,
+        linear_equiv.coe_coe, linear_map.coe_comp, tensor_product.map_tmul,
+        tensor_product.comm_two_in_four_tmul], },
+    { intros z₁ z₂ h₁ h₂,
+      simp only [h₁, h₂, smul_add, lie_add], }, },
+  { intros z₁ z₂ h₁ h₂,
+    simp only [h₁, h₂, smul_add, add_lie], },
+end
+
+instance lie_algebra : lie_algebra A (A ⊗[R] L) :=
+{ lie_smul := bracket_lie_smul R A L, }
+
+end extend_scalars
+
+namespace restrict_scalars
+
+open restrict_scalars
+
+variables [h : lie_ring L]
+
+include h
+
+instance : lie_ring (restrict_scalars R A L) := h
+
+variables [comm_ring A] [lie_algebra A L]
+
+@[simp] lemma linear_equiv_map_lie (x y : restrict_scalars R A L) :
+  linear_equiv R A L ⁅x, y⁆ = ⁅linear_equiv R A L x, linear_equiv R A L y⁆ :=
+rfl
+
+@[nolint unused_arguments]
+instance lie_algebra [comm_ring R] [algebra R A] : lie_algebra R (restrict_scalars R A L) :=
+{ lie_smul := λ t x y, by { rw ← (linear_equiv R A L).to_equiv.apply_eq_iff_eq, simp, },
+  .. (infer_instance : module R (restrict_scalars R A L)), }
```

Comment:
You could also write this
```suggestion
{ lie_smul := λ t x y, (lie_smul _ (show L, from x) (show L, from y) : _),
  .. (by apply_instance : module R (restrict_scalars R A L)), }
```
which eliminates the need for `linear_equiv`.

Reply:
Personally I slightly prefer the version using `linear_equiv` since it does not rely on the fact that `restrict_scalars R A L` is definitionally just `L`. Nevertheless I don't feel strongly and I freely admit I am inconsistent since I use the definitional equivalence to define the `lie_ring` structure just 10 lines higher up.

Incidentally I note you also switched `infer_instance` to `by apply_instance`. I saw some remarks (of Floris I think, but I can't find them now) that basically we should always use `by apply_instance` so maybe I'll start doing that from now on.